ql/src/test/results/clientpositive/llap/acid_no_buckets.q.out

```diff
@@ -810,18 +810,18 @@ STAGE PLANS:
                       Statistics: Num rows: 500 Data size: 265500 Basic stats: COMPLETE Column stats: PARTIAL
                       Reduce Output Operator
                         key expressions: _col0 (type: struct<writeid:bigint,bucketid:int,rowid:bigint>)
-                        null sort order: z
+                        null sort order: a
```

Comment:
I don't think it will have any impact (row_id should not be null) but why is null order changing? I do not see any related change in your changes?

Reply:
Null order is not depending the setting `hive.default.nulls.last`:
https://github.com/apache/hive/blob/02be85202eb46f7207ce2925b361425d87638973/ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java#L6936

I tried to tackle this in #2227 but I experienced some regression in SMB joins: result set mismatch because rows with nulls introduced.

Let me check how was this handled prior this patch: I think the null order came from the parent RS (which introduce the sort by ROWID) not the child (enforce bucketing + sort by ROWID).

Reply:
I found that SMB join expects nulls first behavior.